### ISSUE TITLE
Unable to build custom nodes from n8n-nodes-starter due to eslint 404

### ISSUE THREAD

**Issue by dylan-bartos-tanium** [n8n-io/n8n#4142]
**Describe the bug**
When using npm link to connect the n8n global install and the n8n-nodes-start project, you receive a 404. The workflow is described in this stream from n8n: https://www.youtube.com/watch?v=OI6zHJ56eW0 from roughly 4-9 minutes. It looks like this was brought about in PR 4050 (https://github.com/n8n-io/n8n/pull/4050)

npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@n8n_io%2feslint-config - Not found
npm ERR! 404 
npm ERR! 404  '@n8n_io/eslint-config@latest' is not in this registry.

**To Reproduce**

1. Clone the n8n-nodes-starter project
2. Install n8n globally (npm install n8n -g)
3. From the starter project directory, run (npm link)
4. From your global install directory, ie /usr/local/lib/node_modules, go to the n8n subdirectory and link your starter project (npm link n8n-nodes-<yourprojectname>

**Expected behavior**
The link appears to be referencing a package which has not been published (@n8n_io/eslint-config@latest). On the n8n project page (https://www.npmjs.com/package/n8n) the eslint project is listed, but it directs to a 404. 

**Environment (please complete the following information):**

- OS: MacOS 12.5.1
- n8n 0.194.0
- Node.js Version 18.9.0
- Database system SQLite
- Operation mode own

**Comment by dylan-bartos-tanium** [comment]
Workaround: If you modify the package.json of your globally installed n8n instance to remove the @n8n_io reference on eslint, you can start linking again.

"@n8n_io/eslint-config": "",

"eslint-config":"",

**Comment by pemontto** [comment]
I build my own docker images to use queue mode. This just started occurring when I updated the tag from `0.193.5` to `0.194.0`.

Weirdly the build fails with this Dockerfile
```dockerfile
ARG N8N_VERSION=latest

FROM n8nio/n8n:${N8N_VERSION}

RUN apk add --no-cache zip git

RUN cd /usr/local/lib/node_modules/n8n \
    && npm install n8n-nodes-mjml n8n-nodes-defang n8n-nodes-ldap n8n-nodes-greynoise n8n-nodes-yaml
```

But succeeds with this:
```dockerfile
ARG N8N_VERSION=latest

FROM n8nio/n8n:${N8N_VERSION}

RUN apk add --no-cache zip git

RUN cd /usr/local/lib/node_modules/n8n
RUN npm install n8n-nodes-mjml
RUN npm install n8n-nodes-defang
RUN npm install n8n-nodes-ldap
RUN npm install n8n-nodes-greynoise
RUN npm install n8n-nodes-yaml
```

**Comment by pemontto** [comment]
Just did a fresh local install and I can't install modules from the cli

```bash
~ on  main [?] 
❯ npm uninstall n8n --location=global

~ on  main [?] 
❯ npm install n8n --location=global  
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.

added 1139 packages, and audited 1140 packages in 45s

106 packages are looking for funding
  run `npm fund` for details

5 high severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

~ on  main [?] 
❯ cd /Users/test/.nvm/versions/node/v16.15.1/lib/node_modules/n8n

lib/node_modules/n8n on  main [?] is 📦 v0.194.0 via  v16.15.1 
❯ npm install n8n-nodes-starter            
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@n8n_io%2feslint-config - Not found
npm ERR! 404 
npm ERR! 404  '@n8n_io/eslint-config@latest' is not in this registry.
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/test/.npm/_logs/2022-09-20T11_04_32_755Z-debug-0.log
```

**Comment by netroy** [comment]
I'm not sure how I feel about installing package inside another installed package (which most definitely will mess with how npm resolutions work); but if it's necessary, please always use `--omit-dev`. 
If you find this as a recommended way of loading custom nodes in n8n in some docs, please send me the links, so that we can update them accordingly.

To address the main issue here: [we have published the dev-dependency to npm ](https://www.npmjs.com/package/@n8n_io/eslint-config).

@dylan-bartos-tanium can you please check if this resolves the issue for you :crossed_fingers: ?

**Comment by pemontto** [comment]
@netroy is there a better way we can build our docker images (or not build them at all), e.g. by mounting our own `~/.n8n/custom/` where we npm install community packages?

**Comment by netroy** [comment]
@pemontto if you are self-hosting n8n, you can go to `/settings/community-nodes` and install the packages directly from there.
The issue with mounting packages from your local filesystem is that if you are running a docker container inside a VM (like in docker for osx and windows), packages built on the host might not work inside the container. 
If your packages aren't published on npm, then you can definitely mount your own `~/.n8n/custom` as an alternative.

**Comment by pemontto** [comment]
@netroy would love to do that, but the feature is unavailable if `EXECUTIONS_MODE=queue`:

![SCR-20220920-ltx](https://user-images.githubusercontent.com/939704/191288882-46b58e01-f4ef-4b75-a193-2f446530e9fe.png)

**Comment by dylan-bartos-tanium** [comment]
@netroy I installed n8n onto my system again to get a fresh copy of the modified package.json file. Then I tried to link the starter repo to the n8n install and it worked without a problem. Sounds like your publish to the registry fixed the 404.

I presume that you'd recommend creating a symlink from the ~/.n8n/custom directory to the n8n-nodes-starter/nodes directory. Then, run npm build on every iteration of the node development and when n8n start is run, the new nodes will be available? I'll try this. 

I searched through the documentation and found the following pages referencing this method of linking, in addition to the livestream mentioned in the original post. 

https://docs.n8n.io/integrations/creating-nodes/test/run-node-locally/
https://docs.n8n.io/integrations/creating-nodes/build/declarative-style-node/#test-your-node
https://docs.n8n.io/integrations/creating-nodes/build/programmatic-style-node/#step-7-update-the-npm-package-details
